odoo/addons/base/models/ir_actions_report.py

```diff
@@ -963,6 +963,8 @@ def _get_rendering_context(self, docids, data):
                 'docs': docs,
             })
         data['is_html_empty'] = is_html_empty
+        # Replaces unescaped(!) headings (h[1-6]) with <p class='h[1-6]'> tags.
+        data['replace_headings'] = lambda x: Markup(re.sub(r"<h(\d)>(.+?)<\/h\1>", "<p class='h\\1'>\\2</p>", str(x), re.M))
```

Comment:
should be `[1-6]` instead of `\d` if we want H1-H6, there is not much difference

should be insensitive to case (`re.I` flag) even if the editor will have it in lower case

this doesn't allow attribute on h1 (maybe there is not in the editor but we could imagine content in a HTML not from the editor)

we can put a P tag in a H1-H7 tag, but we can't put a P tag in a P tag (eg. browser renderer will correct `<p id="hello"><p>world</p></p>` to `<p id="hello"></p><p>world</p><p></p>`), so if we only fixed H1 maybe a better solution would be to replace H1 by H2. If doing that you could do a simpler regex, only targeting `<h(/?)1\b`, this could even just be a replace without regex (with 2 replace for lower case or 4 with upper case tags).

the regex usually are often compiled one time in the top of the file and referenced (instead of compiled at each function call), but it doesn't seem too important here.